docs/datastore-emulator-setup.md

```diff
@@ -0,0 +1,27 @@
+# Datastore emulator setup
+
+## Why we need an emulator?
+
+The Java data access API ([Objectify](https://github.com/objectify/objectify)) used for TEAMMATES will be upgraded from v5 to v6.
+
+The latest version (v6+) uses `Cloud Datastore API` instead of `Google App Engine API` used in previous versions.
+
+In order to locally simulate production Datastore environment during development and testing of relevant features, we will rely on this tool called Datastore Emulator.
+
+## Steps to local setup
+Refer to the official [installation guide](https://cloud.google.com/datastore/docs/tools/datastore-emulator).
+
+## Troubleshoot
+1. Sometimes the recommended emulator setup in [wiki](https://github.com/objectify/objectify/wiki/Setup#initialising-the-objectifyservice-to-work-with-emulator-applies-to-v6) might not work with `Exiting due to exception: java.io.IOException: Failed to bind`.
+
+**Solution:**
+In that case, try out some other localhost ports such as `gcloud beta emulators datastore start --host-port=localhost:8482`.
+Also change `DatastoreOptions.setHost()` parameter in `src/main/java/teammates/storage/api/OfyHelper.java` accordingly.
```

Comment:
The error is due to the emulator not shutting down properly, which leaves a process running in `port 8484`. Since we have fixed the port number in `build.template.properties` to `port 8484`, I guess we should change this part?

To kill the dangling process in `port 8484` we can find its port id with the command `lsof -i tcp:8484`, and kill it with `kill -9 <PID>`

Reply:
@jianhandev Right, you can go ahead and co-author on this.

Reply:
Alright, will add in some other details also and seek your review afterwards.